include/picongpu/plugins/openPMD/openPMDWriter.hpp

```diff
@@ -860,11 +860,16 @@ make sure that environment variable OPENPMD_BP_BACKEND is not set to ADIOS1.
                 ::openPMD::MeshRecordComponent mrc = mesh[::openPMD::RecordComponent::SCALAR];
                 std::string datasetName = params->openPMDSeries->meshesPath() + name;
 
+                auto numRNGsPerSuperCell = DataSpace<simDim>::create(1);
+                numRNGsPerSuperCell.x() = numFrameSlots;
                 // rng states are always of the domain size therefore query sizes from domain information
                 const SubGrid<simDim>& subGrid = Environment<simDim>::get().SubGrid();
-                pmacc::math::UInt64<simDim> recordLocalSizeDims = subGrid.getLocalDomain().size;
-                pmacc::math::UInt64<simDim> recordOffsetDims = subGrid.getLocalDomain().offset;
-                pmacc::math::UInt64<simDim> recordGlobalSizeDims = subGrid.getGlobalDomain().size;
+                pmacc::math::UInt64<simDim> recordLocalSizeDims{
```

Comment:
This is intended, we have now N-RNG states per supercell and not supercell cells RNG states per supercell.
N-is the number of slots in a frame because the number of slots is >= number of cells per supercell